docs: remove needless prettier-ignore

otegami opened the pull request:
GitHub: https://github.com/groonga/groonga/pull/2271#discussion_r2034482549

It seems that recent MyST-Parser accepts the following Prettier style format as reference directive(?):

```markdown
(XXX)=

# XXX
```

So we can remove these `prettier-ignore`.

kou commented:
> We don't use prettier in Groonga documentation now.

Really!?

otegami commented:
> Really!?

I didn't use it.. Do we use it?
If so, I couldn't check it the previous https://github.com/groonga/groonga/pull/2271#discussion_r2034409609.

kou commented:
We're using Prettier via pre-commit: https://github.com/groonga/groonga/blob/58bffdca304895d8d49fd5b6f5356ed03bae7cbc/.pre-commit-config.yaml#L188-L196

Our CI always runs it: https://github.com/groonga/groonga/blob/58bffdca304895d8d49fd5b6f5356ed03bae7cbc/.github/workflows/lint.yml#L24

otegami commented:
Thanks you so much. I misunderstood where Prettier works.
I've update the description and I confirmed there are no errors as follows.

```console
$ git reset --soft HEAD~
$ git status
On branch remove-prettier-settings
Changes to be committed:
  (use "git restore --staged <file>..." to unstage)
	modified:   doc/source/contribution/documentation/introduction.md
	modified:   doc/source/news/13.md
	modified:   doc/source/news/14.md
	modified:   doc/source/news/15.md
$ pre-commit run --all-files
clang-format.............................................................Passed
cmake-format.............................................................Passed
prettier.................................................................Passed
```

kou commented:
> Prettier doesn't raise error for reference sections, so we can remove these prettier-ignore.

Prettier uses

```markdown
(XXX)=

# XXX
```

format not

```markdown
(XXX)=
# XXX
```

. It's not changed.

Old (?) MyST-Parser doesn't accept 

```markdown
(XXX)=

# XXX
```

. So I disabled the Preitter for the part. But recent (?) MyST-Parser accepts both of

```markdown
(XXX)=
# XXX
```

and

```markdown
(XXX)=

# XXX
```

. So we can remove the `prettier-ignore` directive.
